src/Build/BackEnd/Components/RequestBuilder/TaskHost.cs

```diff
@@ -989,5 +991,45 @@ private void VerifyActiveProxy()
         {
             ErrorUtilities.VerifyThrow(_activeProxy, "Attempted to use an inactive task host.");
         }
+
+        /// <summary>
+        /// Inialize new RAR node
+        /// </summary>
+        bool IRarBuildEngine.CreateRarNode()
+        {
+            var nodeManager = _host.GetComponent(BuildComponentType.NodeManager) as INodeManager;
+
+            var configuration = new NodeConfiguration
+            (
+                -1, /* must be assigned by the NodeManager */
+                _host.BuildParameters,
+                null
+    #if FEATURE_APPDOMAIN
+                , AppDomain.CurrentDomain.SetupInformation
+    #endif
+                , new LoggingNodeConfiguration(_host.LoggingService.IncludeEvaluationMetaprojects, _host.LoggingService.IncludeEvaluationProfile, _host.LoggingService.IncludeTaskInputs)
```

Comment:
For this sort of method, it might be better to arrange it so you can call BuildManager rather than duplicating the logic.